packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -109,13 +119,36 @@ abstract class KeyBase extends Resource implements IKey {
    * must not be empty and so default grants won't work.
    */
   public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
-    return iam.Grant.addToPrincipalAndResource({
-      grantee,
-      actions,
-      resourceArns: [this.keyArn],
-      resource: this,
-      resourceSelfArns: ['*']
-    });
+    // KMS verifies whether the principals included in its key policy actually exist.
+    // This is a problem if the stack the grantee is part of depends on the key stack
+    // (as it won't exist before the key policy is attempted to be created).
+    // In that case, make the account the resource policy principal
+    const granteeStackDependsOnKeyStack = this.granteeStackDependsOnKeyStack(grantee);
```

Comment:
Do we have some doc reference (or appsec consultation) that this is the best practice?

Reply:
What would you suggest as an alternative here, given the comment?